components/fxa-client/src/state_machine/internal_machines/mod.rs

```diff
@@ -177,6 +171,66 @@ impl State {
     }
 }
 
+/// Number of times to retry fxa calls in the face of network errors
+const NETWORK_RETRY_LIMIT: usize = 3;
+/// Number of times to retry fxa calls in the face of auth issues
+const AUTH_RETRY_LIMIT: usize = 1;
```

Comment:
I'm interested to know why we bother retrying here at all?

Reply:
Hmm yeah - logically what we would be doing is:
- get a 4xx that tells us our access token has expired,
- use our refresh token to get a new access token,
- try again

Just retrying won't do anything so +1 to make sure we check our refresh token and ask for a new access token before retrying